Expensify/expensify-common
# update inlineCodeBlock replace regex

This PR updated `inlineCodeBlock` replace regex to preventing existence of code fence(`<pre>` tag) inside inline code(`<code>` tag).
So if `<pre>` tag already exists inside backticks(`&#x60;`), it blocks replacement of `inlineCodeBlock`.

Before update:
```
`(```test```)` => <code>(<pre>test</pre>)</code>
```
After update:
```
`(```test```)` => &#x60;(<pre>test</pre>)&#x60;
```

### Fixed Issues
$ [GH_LINK](https://github.com/Expensify/App/issues/26966)

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
I added "Test code fence within inline code" test case for this.
2. What tests did you perform that validates your changed worked?
I ran "npm run test" and confirmed all tests passed successfully.

# QA
1. What does QA need to do to validate your changes?
Open any chat.
Write "`(```test```)`" in the composer and send the message.
The message is shown immediately in expected format.
```
`
test
`
```

2. What areas to they need to test for regressions?
Run "npm run test".
